### ISSUE TITLE
Implement `BrowserType.connect(wsEndpoint[, options])`

### ISSUE THREAD

**Issue by robingustafsson** [grafana/xk6-browser#17]
Add support for `BrowserType.connect(wsEndpoint[, options])`, to connect to an existing browser instance at specified CDP WS endpoint.

**Relevant links**:
- Playwright docs: https://playwright.dev/docs/api/class-browsertype#browser-type-connect
- xk6-browser code starting point: https://github.com/grafana/xk6-browser/blob/main/chromium/browser_type.go#L56
- Playwright code: https://github.com/microsoft/playwright/blob/master/packages/playwright-core/src/server/chromium/crBrowser.ts#L50
- Playwright tests: https://github.com/microsoft/playwright/blob/master/tests/browsertype-connect.spec.ts

**Comment by inancgumus** [comment]
I think this is a good next issue to work on because it's good to keep the browser open and try it with updated code while fixing bugs. So my workflow becomes faster.

**Comment by inancgumus** [comment]
I've been researching about this feature, and I have a few things to clarify before I start working on this :)

So I appreciate your feedback 🙇 ❤️ 
@robingustafsson @imiric 

---

#### 1st Question

[Playwright.Connect method](https://playwright.dev/docs/api/class-browsertype#browser-type-connect) takes a `wsEndpoint`:

```javascript
browserType.connect(wsEndpoint[, options])​ <Promise<Browser>>
    wsEndpoint <string> A browser websocket endpoint to connect to.
    ...

// This method attaches Playwright to an existing browser instance.
```

So we probably need to change the `Connect` method in `API/browser_type.go` from this:

https://github.com/grafana/xk6-browser/blob/758d14bb6082e0c8bbea7933522f7f84cff985fc/api/browser_type.go#L29

To this:

```golang
Connect(wsEndpoint string, opts goja.Value)
```

---

#### 2nd Question

We probably need to add a new method to [the module's API](https://github.com/grafana/xk6-browser/blob/main/main.go) like this:

```golang
func (m *JSModule) Connect(wsEndpoint string, opts goja.Value) api.Browser {
	// k6common.Throw(m.vu.Runtime(), errors.New("In Progress 🚀"))
	if browserName == "chromium" {
		bt := chromium.NewBrowserType(m.vu.Context())
		return bt.Connect(wsEndpoint, opts)
	}
	...
}
```

And then call it like this:

```javascript
import chromium from 'k6/x/browser';

export default function() {
  const browser = chromium.connect("wss://...", { ... });
  const context = browser.newContext();
  const page = context.newPage();
  ...
}
```

---

#### 3rd Question

It seems like Playwright's `Connect` method connects to a Playwright server (IDK what it is 🤷—As far as I see, it might be a way to abstract browsers somehow), but we don't have a server to connect to. They also have a `ConnectOverCDP` method.

[This comment](https://github.com/microsoft/playwright/issues/1985#issuecomment-901124421) tells the difference:

> browserType.connectOverCDP = uses Chrome CDP protocol and works by that only with Chromium-based browsers
> browserType.connect = uses Playwright own protocol and works with all Playwright browsers

Since we currently only support Chrome, we will need to adjust the `common/Browser` type to use the WebSocket URL and give it here (_instead of getting it from a browser process_). So the rest of the things will be the same:

https://github.com/grafana/xk6-browser/blob/758d14bb6082e0c8bbea7933522f7f84cff985fc/common/browser.go#L104-L106

And here:

https://github.com/grafana/xk6-browser/blob/758d14bb6082e0c8bbea7933522f7f84cff985fc/common/browser.go#L110-L113

Then, we'll need to skip tracking the following when the `Browser` value is created by the `Connect` method:

https://github.com/grafana/xk6-browser/blob/758d14bb6082e0c8bbea7933522f7f84cff985fc/common/browser.go#L176

Are we going to support closing the browser over CDP? As in here:

https://github.com/grafana/xk6-browser/blob/758d14bb6082e0c8bbea7933522f7f84cff985fc/common/browser.go#L341-L359

Even if we want to do so, I think we'll need to skip calling these:

https://github.com/grafana/xk6-browser/blob/758d14bb6082e0c8bbea7933522f7f84cff985fc/common/browser.go#L349-L350

---

In summary, the plan is to give WebSocket URL to a new `Browser` and let it do the rest.

**Comment by robingustafsson** [comment]
1. Yes, the method signature is wrong in the code 🙂
2. Hm, yeah this is why I wanted to expose the `chromium` `BrowserType`: https://github.com/grafana/xk6-browser/blob/1e901b4a803bb932a4efe60380d85ee3dec9209f/main.go#L40
3. Yes, we don't have a client/server model as Playwright does (to support multiple client-side language SDKs), so only need to support connecting over CDP to a running browser. I think we should break PW compat here an use `BrowserType.connect()` for connecting over CDP.

**Comment by inancgumus** [comment]
We might want to resurrect this issue if we want to connect to a browser running in a Docker container. Also, the `connect` method can automatically use this if the extension runs on the Cloud. Moving this to the v0.9.0 milestone until we decide not to include it.
